dispatch/apps/api/views.py

```diff
@@ -407,6 +407,42 @@ def articles(self, request, pk=None):
         view = ArticleViewSet.as_view({'get': 'topic'})
         return view(request, pk=pk)
 
+class FileViewSet(viewsets.ModelViewSet):
+    """
+    Viewset for File model views.
+    """
+    model = File
+    serializer_class = FileSerializer
+
+    def get_queryset(self):
+        queryset = File.objects.all()
+        q = self.request.query_params.get('q', None)
+        if q is not None:
+            # If a search term (q) is present, filter queryset by term against `name`
+            queryset = queryset.filter(name__icontains=q)
+        return queryset
+
+    def bulk_delete(self, request):
+        deleted = []
+        ids = self.request.data.get('ids', None)
```

Comment:
I think the motivation behind the separate queries was to catch errors on individual files (e.g. trying to delete a file that doesn't exist). However, this code doesn't actually have any proper error handling (not your fault, @JamieRL).

This method kinda falls outside of the standard REST architecture (not to mention the HTTP protocol), so I'm not sure what to do in the case where one or more of the files can't be deleted. Instinct tells me we should return some sort of error, but I'm unsure of the format. Do we return a 400 if only some of the operations fail? 

I think it makes the most sense to return an error if any of the operations fail; since the user requested to delete _all_ of the ids, it doesn't make sense to return 200 if that wasn't the case.

If we want to get really fancy we could use status code 207: http://www.restpatterns.org/HTTP_Status_Codes/207_-_Multi-Status

However, I'm leaning towards just scrapping the batch endpoint and refactoring the frontend to make multiple calls to the `article/{id}/delete` route. 

@blx @JamieRL thoughts?